ngraph/core/include/ngraph/visibility.hpp

```diff
@@ -16,7 +16,7 @@
 
 // https://gcc.gnu.org/wiki/Visibility
 // Generic helper definitions for shared library support
-#if defined _WIN32 || defined __CYGWIN__
+#if (defined _WIN32 || defined __CYGWIN__) && !defined(NGRAPH_STATIC_LIBRARY)
```

Comment:
IE and ngraph declarations look different. In case of static library IE does not export anything, while ngraph does for UNIX.

Reply:
We have one more macro `TRANSFORMATIONS_API`, looks like we need to fix it as well.

Reply:
TRANSFORMATIONS_API uses NGRAPH_HELPER_DLL_IMPORT and NGRAPH_HELPER_DLL_EXPORT that are covered by the static check

`#ifdef inference_engine_transformations_EXPORTS
#define TRANSFORMATIONS_API NGRAPH_HELPER_DLL_EXPORT
#else
#define TRANSFORMATIONS_API NGRAPH_HELPER_DLL_IMPORT
#endif

#if (defined _WIN32 || defined __CYGWIN__) && !defined(NGRAPH_STATIC_LIBRARY)
#define NGRAPH_HELPER_DLL_IMPORT __declspec(dllimport)
#define NGRAPH_HELPER_DLL_EXPORT __declspec(dllexport)
#define NGRAPH_HELPER_DLL_LOCAL
#elif defined(__GNUC__) && __GNUC__ >= 4
#define NGRAPH_HELPER_DLL_IMPORT __attribute__((visibility("default")))
#define NGRAPH_HELPER_DLL_EXPORT __attribute__((visibility("default")))
#define NGRAPH_HELPER_DLL_LOCAL __attribute__((visibility("hidden")))
#else
#define NGRAPH_HELPER_DLL_IMPORT
#define NGRAPH_HELPER_DLL_EXPORT
#define NGRAPH_HELPER_DLL_LOCAL
#endif
`